### PR TITLE
Gateway tests

### PR DESCRIPTION
**Pull request by ben-d758**
# What
- Add unit test to case notes and resident gateway
- Update mapping of fields resident gateway 

# Why
- To add more confidence in the gateways in the future
- to remove unused mapping fields for front end

# Notes
- still incomplete but merging this PR now as we are largely out of sync with master

Link to ticket
https://trello.com/c/2FiufpNn/16-add-unit-tests-for-frontend-gateways

### CODE REVIEW

File: src/gateways/case-notes.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    beforeEach(() => {
+        moxios.install();
+    });
+
+    afterEach(() => {
+        moxios.uninstall();
+    });
+
+    it('200 Success "/residents/:residendId/caseNotes"', async () => {
```
Comment (NyWeb) [inline src/gateways/case-notes.test.js:15]: I think, think more carefully about test names, slightly more readable would be nice.

Like Returns status 200 when... - the URL isn't important in the test name

**Review by NyWeb — APPROVED** [review]
I think it looks good, but have another think about test names perhaps?

File: src/gateways/case-notes.test.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            {
+                Id: 2,
+                CaseNote: 'vero qui voluptas',
+                HelpRequestId: 16,
+                ResidentId: randomId,
+                CreatedAt: '2021-01-20T05:06:34.567Z'
+            }
+        ];
+
+        const expectedResponse = [
```
Comment (kosiakkatrina) [inline src/gateways/case-notes.test.js:35]: Could this just be `expectedResponse = mockAxiosResponse` 🤔

Comment (kosiakkatrina) [reply src/gateways/case-notes.test.js:35]: Nevermind, just saw the casing

**Review by kosiakkatrina — COMMENTED** [review]
I know it's WIP, but it might be worth writing some tests for when we get an error from the API and also maybe unit tests for mapping?

**Comment by ben-d758** [comment]
> I know it's WIP, but it might be worth writing some tests for when we get an error from the API and also maybe unit tests for mapping.
> Nice work tho!

So the `getResident` method in the `ResidentGateway` is responsible for mapping the response so this test would check that that works as it should. Or is that not clear and does it need to be separated out?

File: src/gateways/case-notes.test.js (hunk 2 above)
Comment (ben-d758) [reply src/gateways/case-notes.test.js:35]: It looks the same but The `mockAxiosResponse` comes back as Pascal Case and that gets changed to camelCase in the `ToResident` mapper so would need to reflect this in the test

File: src/gateways/case-notes.test.js (hunk 1 above)
Comment (ben-d758) [reply src/gateways/case-notes.test.js:15]: Thanks for pointing that out, will update

**Pull request merged**